canvas_sdk/protocols/clinical_quality_measure.py

```diff
@@ -41,3 +52,52 @@ def protocol_key(cls) -> str:
         External key used to identify the protocol.
         """
         return cls.__name__
+
+    @property
+    def timeframe(self) -> Timeframe:
+        """The default Timeframe (self.timeframe) for all protocols.
+        This defaults to have a start of 1 year ago and an end time of the current time.
+        Plugin authors can override this if a different timeframe is desired.
+        """
+        end = self.now
+        return Timeframe(start=end.shift(years=-1), end=end)
+
+    @property
+    def now(self) -> arrow.Arrow:
+        """A convenience method for returning the current datetime."""
+        return arrow.utcnow()
```

Comment:
Do we want this calculated every time it is called?

I have a similar change here, inspired by the workflow kit: https://github.com/canvas-medical/canvas-plugins/pull/194/files

Reply:
That's a good call. I'll change it.

Reply:
Done.